geom/geojson_unmarshal.go

```diff
@@ -68,6 +34,42 @@ func UnmarshalGeoJSON(input []byte, nv ...NoValidate) (Geometry, error) {
 	return g, nil
 }
 
+func chooseGeoJSONCoordinatesType(hasLength map[int]bool) CoordinatesType {
+	// The GeoJSON spec allows (but doesn't require) parsers to ignore any
+	// "extra" coordinate values in addition to the normal 3 coordinate values
+	// used to specify a 3D position. We choose to interpret a 4th dimension as
+	// M values, and ignore any further (5th and above) dimensions without
+	// error.
+	//
+	// We choose the highest ctype such that we can unambiguously parse any of
+	// the points in the geometry (i.e. we don't have to make up a Z or M
+	// value).
+	//
+	// | has2  | has3  | has4OrMore | ctype   |
+	// | ---   | ---   | ---        | ---     |
+	// | false | false | false      | DimXY   |
+	// | false | false | true       | DimXYZM |
+	// | false | true  | false      | DimXYZ  |
+	// | false | true  | true       | DimXYZ  |
+	// | true  | false | false      | DimXY   |
+	// | true  | false | true       | DimXY   |
+	// | true  | true  | false      | DimXY   |
+	// | true  | true  | true       | DimXY   |
+	var has2, has3, has4OrMore bool
+	for length := range hasLength {
```

Comment:
`hasLength` is a "set", but we're not checking the bool here. Would be better to use a `map[int]struct{}` instead to make it less ambiguous.